lib/jxl/encode.cc

```diff
@@ -387,17 +387,20 @@ JxlEncoderStatus JxlEncoderStruct::RefillOutputByteQueue() {
       ib.origin.x0 = input_frame->option_values.header.layer_info.crop_x0;
       ib.origin.y0 = input_frame->option_values.header.layer_info.crop_y0;
     }
+    JXL_ASSERT(writer.BitsWritten() == 0);
     if (!jxl::EncodeFrame(input_frame->option_values.cparams, frame_info,
                           &metadata, input_frame->frame, &enc_state, cms,
                           thread_pool.get(), &writer,
                           /*aux_out=*/nullptr)) {
       return JXL_API_ERROR("Failed to encode frame");
     }
+    codestream_bytes_written_beginning_of_frame =
```

Comment:
As far as I can tell, it's now only counting the bytes produced by EncodeFrame, but not including the ones produced by WriteHeaders as well as WriteICC higher up, so updating the codestream bytes pos is also needed there.

I think that should be all (anything that outputs jxlc or jxlp), but it may be worth double checking. A preview frame also goes to the codestream should also be counted, but preview image is not supported yet by the encoder so that can't be done yet.

Reply:
added